lib/cppcheck.cpp

```diff
@@ -1352,6 +1352,79 @@ void CppCheck::getErrorMessages()
     Preprocessor::getErrorMessages(this, &s);
 }
 
+void CppCheck::analyseClangTidy(const ImportProject::FileSettings &fileSettings )
+{
+    std::string allIncludes = "";
+    std::string allDefines = "-D"+fileSettings.defines;
```

Comment:
Do you use `defines` instead of `cppcheckDefines()` intentionally here?
For the normal check `cppcheckDefines()` is used, I really do not know, but maybe it should be used here too?

Reply:
for the clang-tidy cli call you have to set the VS preprocessor defines and not the cppcheck ones. Afaik fileSettings.defines are those. So yes I used them intentionally here ;)